docs/proposals/scheduledjob.md

```diff
@@ -0,0 +1,222 @@
+and the schedule in ISO8601 format.  This implies that each ScheduledJob execution
+will be created from the JobSpec actual at a point in time when the execution will
+be started.  This also implies that any changes to ScheduledJobSpec will be applied
+upon subsequent execution of a job.
+
+```go
+// ScheduledJobSpec describes how the job execution will look like and when it will actually run.
+type ScheduledJobSpec struct {
+
+    // Schedule contains the schedule in ISO8601 format, eg.
+    // - 2015-07-21T14:00:00Z - represents date and time in UTC
+    // - R/2015-07-21T14:00:00Z/P1D - represents endlessly repeating interval (1 day), starting from given date
+    Schedule string
+
+    // Optional duration in seconds the job will still run if it misses scheduled
+    // time for any reason.  Missed jobs executions will be counted as failed ones.
+    Epsilon *int64
+
+    // ConcurrencyPolicy specifies how to treat concurrent executions of a Job.
+    ConcurrencyPolicy ConcurrencyPolicy
+
+    // Suspend flag tells the controller to suspend subsequent executions, it does
+    // not apply to already started executions.  Defaults to false.
+    Suspend bool
+
+    // Template is the object that describes the job that will be created when
+    // executing a ScheduledJob.
+    JobTemplate JobSpec
+
+    // Key of the selector that is added to Jobs to prevent concurrently running
+    // Jobs selecting their pods.
+    // Users can set this to an empty string to indicate that the system should
+    // not add any selector and label. If unspecified, system uses
+    // "scheduledjob.kubernetes.io/podTemplateHash".
+    // Value of this key is hash of ScheduledJobSpec.PodTemplateSpec.
+    // No label is added if this is set to empty string.
+    UniqueLabelKey string
+}
+
+// ConcurrencyPolicy describes how the job will be handled.
+// Only one of the following concurrent policies may be specified.
+// If none of the following policies is specified, the default one
+// is AllowConcurrent.
+type ConcurrencyPolicy string
+
+const (
+    // AllowConcurrent allows ScheduledJobs to run concurrently.
+    AllowConcurrent ConcurrencyPolicy = "Allow"
+
+    // ForbidConcurrent forbids concurrent runs, skipping next run if previous
+    // hasn't finished yet.
+    ForbidConcurrent ConcurrencyPolicy = "Forbid"
+
+    // ReplaceConcurrent cancels currently running job and replaces it with a new one.
+    ReplaceConcurrent ConcurrencyPolicy = "Replace"
+)
+```
+
+`ScheduledJobStatus` structure is defined to contain information about scheduled
+job executions.  The structure holds a list of currently running job instances
+and additional information about overall successful and unsuccessful job executions.
+
+```go
+// ScheduledJobStatus represents the current state of a Job.
+type ScheduledJobStatus struct {
+    Conditions []ScheduledJobCondition
+
+    // Successful tracks the overall amount of successful completions of this job.
```

Comment:
You've raised that issue already when implementing JobController. I'm thinking about taking different approach here. Instead of counting pods each time I'm thinking about updating this value with new information. Not sure how to implement it yet, though ;) Still the general idea of having those tracked should stay.